### ISSUE TITLE
The training of Stable-diffusion-upscale-4x model

### ISSUE THREAD

**Issue by WUMINGCHAzero** [huggingface/diffusers#3185]
I try to fine-tune the Stable-diffusion-upscale-4x model on custom data, but get very wired results, much worse than original one. Can anyone help? LoRA is used and part of my training code is as follows (input image is already normalized to [-1, 1]):
```python
for epoch in range(first_epoch, args.num_train_epochs):
        unet.train()
        train_loss = 0.0
        for step, batch in enumerate(train_dataloader):
            # Skip steps until we reach the resumed step
            if args.resume_from_checkpoint and epoch == first_epoch and step < resume_step:
                if step % args.gradient_accumulation_steps == 0:
                    progress_bar.update(1)
                continue

            with accelerator.accumulate(unet):
                # Convert images to latent space
                for k in batch:
                    if 'values' in k:
                        batch[k] = get_input(batch, k)
                
                latents = vae.encode(batch["pixel_values"]).latent_dist.sample()
                latents = latents.to(dtype=weight_dtype)
                latents = latents * vae.config.scaling_factor

                # Sample noise that we'll add to the latents
                noise = torch.randn_like(latents)
                if args.noise_offset:
                    # https://www.crosslabs.org//blog/diffusion-with-offset-noise
                    noise += args.noise_offset * torch.randn(
                        (latents.shape[0], latents.shape[1], 1, 1), device=latents.device
                    )

                bsz = latents.shape[0]
                # Sample a random timestep for each image
                timesteps = torch.randint(0, noise_scheduler.config.num_train_timesteps, (bsz,), device=latents.device)
                timesteps = timesteps.long()

                # Add noise to the latents according to the noise magnitude at each timestep
                # (this is the forward diffusion process)
                noisy_latents = noise_scheduler.add_noise(latents, noise, timesteps)

                # add for sd_upscaling
                noise_level = torch.tensor([20], dtype=torch.long, device=noisy_latents.device)
                image_lr = batch['pixel_values_rendered'].to(noisy_latents)
                noise_lr = torch.randn_like(image_lr)
                image_lr = low_res_scheduler.add_noise(image_lr, noise_lr, noise_level)
                latent_model_input = torch.cat([noisy_latents, image_lr], dim=1)
                noise_level = torch.cat([noise_level]*latent_model_input.shape[0])

                # Get the text embedding for conditioning
                encoder_hidden_states = text_encoder(batch["input_ids"])[0]

                # Get the target for loss depending on the prediction type
                if noise_scheduler.config.prediction_type == "epsilon":
                    target = noise
                elif noise_scheduler.config.prediction_type == "v_prediction":
                    target = noise_scheduler.get_velocity(latents, noise, timesteps)
                else:
                    raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")

                # Predict the noise residual and compute loss
                model_pred = unet(latent_model_input, timesteps, encoder_hidden_states, noise_level).sample
                loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")

                # Gather the losses across all processes for logging (if we use distributed training).
                avg_loss = accelerator.gather(loss.repeat(args.train_batch_size)).mean()
                train_loss += avg_loss.item() / args.gradient_accumulation_steps

                # Backpropagate
                accelerator.backward(loss)
                if accelerator.sync_gradients:
                    params_to_clip = lora_layers.parameters()
                    accelerator.clip_grad_norm_(params_to_clip, args.max_grad_norm)
                optimizer.step()
                lr_scheduler.step()
                optimizer.zero_grad()
         
            # Checks if the accelerator has performed an optimization step behind the scenes
            if accelerator.sync_gradients:
                progress_bar.update(1)
                global_step += 1
                accelerator.log({"train_loss": train_loss}, step=global_step)
                train_loss = 0.0

                if global_step % args.checkpointing_steps == 0:
                    if accelerator.is_main_process:
                        save_path = os.path.join(args.output_dir, 'logs', args.tracker_name, f"checkpoint-{global_step}")
                        accelerator.save_state(save_path)
                        logger.info(f"Saved state to {save_path}")

            logs = {"step_loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0]}
            progress_bar.set_postfix(**logs)

            if global_step >= args.max_train_steps:
                break
            
            if accelerator.is_main_process:
                if args.validation_prompt is not None and global_step % args.validation_steps == 0 and step % args.gradient_accumulation_steps == 0:
                    logger.info(
                        f"Running validation... \n Generating {args.num_validation_images} images with prompt:"
                        f" {args.validation_prompt}."
                    )
                    # create pipeline
                    pipeline = StableDiffusionUpscalePipeline.from_pretrained(
                        args.pretrained_model_name_or_path,
                        unet=accelerator.unwrap_model(unet),
                        revision=args.revision,
                        torch_dtype=weight_dtype,
                    )
                    pipeline = pipeline.to(accelerator.device)
                    pipeline.set_progress_bar_config(disable=True)

                    # run inference
                    generator = torch.Generator(device=accelerator.device).manual_seed(args.seed)
                    image_lr = Image.open(args.validation_lr_image).resize((128, 128))

                    images = []
                    for _ in range(args.num_validation_images):
                        images.append(
                            pipeline(args.validation_prompt, image_lr, num_inference_steps=30, generator=generator).images[0]
                        )
                    save_dir = os.path.join(args.output_dir, 'logs', args.tracker_name, 'image_logs')
                    for idx, image in enumerate(images):
                        save_path = os.path.join(save_dir, f'step_{global_step}.png')
                        image.save(save_path)

                    del pipeline
                    torch.cuda.empty_cache()
           
            if accelerator.is_main_process and (global_step!=0) and global_step % args.checkpointing_steps == 0 and step % args.gradient_accumulation_steps == 0:
                unet.save_attn_procs(os.path.join(args.output_dir, 'logs', args.tracker_name, f'global_step_{global_step}'))
```
Many thanks!

**Comment by sayakpaul** [comment]
Interesting! Do you get better results without LoRA? 

What kind of dataset are you using? 

Also, could you shed more light into the kind of result you are getting?

Cc: @yiyixuxu

**Comment by Tangshitao** [comment]
I have a similar problem. Is there a training script for reference?

**Comment by patrickvonplaten** [comment]
We don't have an upscaling script (yet) - maybe also worth asking a question on: https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63

**Comment by oubotong** [comment]
Same issue. Looking for some guidance on finetuning 4x upscale model
